samples/DevTestLabs/Modules/Library/Az.DevTestLabs2.psm1

```diff
@@ -0,0 +1,2636 @@
+    [ValidateNotNullOrEmpty()]
+    [string] $ArtifactRepoBranch = "master",
+
+    [parameter(Mandatory=$false, ValueFromPipelineByPropertyName = $true, HelpMessage="Expiration of announcement (format '2100-01-01T17:00:00+00:00').")]
+    [ValidateNotNullOrEmpty()]
+    [string] $ArtifactRepoFolder = "Artifacts/",
+
+    [parameter(Mandatory=$false, ValueFromPipelineByPropertyName = $true, HelpMessage="Expiration of announcement (format '2100-01-01T17:00:00+00:00').")]
+    [ValidateSet("VsoGit", "GitHub")]
+    [string] $ArtifactRepoType = "GitHub",
+
+    [parameter(Mandatory=$true, ValueFromPipelineByPropertyName = $true, HelpMessage="Expiration of announcement (format '2100-01-01T17:00:00+00:00').")]
+    [ValidateNotNullOrEmpty()]
+    [string] $ArtifactRepoSecurityToken,
+
+    [parameter(Mandatory=$false,HelpMessage="Run the command in a separate job")]
+    [switch] $AsJob = $False
+  )
+
+  begin {. BeginPreamble}
+  process {
+    try {
+      foreach($l in $Lab) {
+        $params = @{
+          labName = $l.Name
+          artifactRepositoryDisplayName = $ArtifactRepositoryDisplayName
+          artifactRepoUri               = $ArtifactRepoUri
+          artifactRepoBranch            = $ArtifactRepoBranch
+          artifactRepoFolder            = $ArtifactRepoFolder
+          artifactRepoType              = $ArtifactRepoType
+          artifactRepoSecurityToken     = $ArtifactRepoSecurityToken
+        }
+        Write-verbose "Set Artifact repo with $(PrintHashtable $params)"
+
+@"
+{
+  "`$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "labName": {
+      "type": "string"
+    },
+    "artifactRepositoryDisplayName": {
+      "type": "string"
+    },
+    "artifactRepoUri": {
+      "type": "string"
+    },
+    "artifactRepoBranch": {
+      "type": "string"
+    },
+    "artifactRepoFolder": {
+      "type": "string"
+    },
+    "artifactRepoType": {
+      "type": "string",
+      "allowedValues": [ "VsoGit", "GitHub" ]
+    },
+    "artifactRepoSecurityToken": {
+      "type": "securestring"
+    }
+  },
+  "variables": {
+    "artifactRepositoryName": "[concat('Repo-', uniqueString(subscription().subscriptionId))]"
+  },
+  "resources": [
+    {
+      "apiVersion": "2018-10-15-preview",
+      "type": "Microsoft.DevTestLab/labs",
+      "name": "[parameters('labName')]",
+      "location": "[resourceGroup().location]",
+      "resources": [
+        {
+          "apiVersion": "2018-10-15-preview",
+          "name": "[variables('artifactRepositoryName')]",
+          "type": "artifactSources",
+          "dependsOn": [
+            "[resourceId('Microsoft.DevTestLab/labs', parameters('labName'))]"
+          ],
+          "properties": {
+            "uri": "[parameters('artifactRepoUri')]",
+            "fold
```

Comment:
I get that `Claim` and `Unclaim` are not allowed PS verbs. It just feel funny calling it `Invoke...` when those for start/stop use the corresponding verbs.

Reply:
Not sure how to fix it. If I use Claim and unclaim the main LINT tools for powershell give an error. My intuition is that people will mostly use the aliases as much simpler, so this might not be too bad ...

Reply:
I understand the predicament. Not sure either. Leave it as is for now.